service/matching/physical_task_queue_manager.go

```diff
@@ -324,6 +324,17 @@ func (c *physicalTaskQueueManagerImpl) Start() {
 	c.logger.Info("", tag.LifeCycleStarted, tag.Cause(c.config.loadCause.String()))
 	c.taggedMetricsHandler.Counter(metrics.TaskQueueStartedCounter.Name()).Record(1)
 	c.partitionMgr.engine.updatePhysicalTaskQueueGauge(c, 1)
+	c.namespaceRegistry.RegisterStateChangeCallback(c.queue, func(ns *namespace.Namespace, deletedFromDb bool) {
+		if deletedFromDb {
+			return
+		}
+
+		if ns.IsGlobalNamespace() &&
+			ns.ReplicationPolicy() == namespace.ReplicationPolicyMultiCluster &&
+			ns.ActiveClusterName() != c.clusterMeta.GetCurrentClusterName() {
+			c.UnloadFromPartitionManager(unloadCauseNamespaceFailover)
```

Comment:
@dnr I am not 100% if I should stop the task queue partition manager.

Reply:
namespace state change callbacks are global, this isn't specific to a particular namespace. i.e. you need one of these at the matching engine level and it should do something like this https://github.com/temporalio/temporal/blob/main/service/matching/matching_engine.go#L281
(except you might not need the double-check and delay)

Reply:
not going to use this approach.